quixstreams/sources/base/source.py

```diff
@@ -320,3 +314,140 @@ def default_topic(self) -> Topic:
 
     def __repr__(self):
         return self.name
+
+
+class StatefulSource(Source):
+    """
+    A `Source` class for custom Sources that need a state.
+
+    Subclasses are responsible for flushing, by calling `flush`, at reasonable intervals.
+
+    Example:
+
+    ```python
+    import random
+    import time
+
+    from quixstreams import Application
+    from quixstreams.sources import StatefulSource
+
+
+    class RandomNumbersSource(StatefulSource):
+        def run(self):
+
+            i = 0
+            state = self.state()
+            while self.running:
+                previous = state.get("number", 0)
+                current = random.randint(0, 100)
+                state.set("number", current)
+
+                serialized = self._producer_topic.serialize(value=current + previous)
+                self.produce(key=str(current), value=serialized.value)
+                time.sleep(0.5)
+
+                # flush the state every 10 messages
+                i += 1
+                if i % 10 == 0:
+                    self.flush()
+                    state = self.state()
+
+
+    def main():
+        app = Application(broker_address="localhost:9092")
+        source = RandomNumbersSource(name="random-source")
+
+        sdf = app.dataframe(source=source)
+        sdf.print(metadata=True)
+
+        app.run()
+
+
+    if __name__ == "__main__":
+        main()
+    ```
+    """
+
+    def __init__(self, name: str, shutdown_timeout: float = 10) -> None:
+        """
+        :param name: The source unique name. It is used to generate the topic configuration.
+        :param shutdown_timeout: Time in second the application waits for the source to gracefully shutdown.
+        """
+        super().__init__(name, shutdown_timeout)
+        self._store_partition: Optional[StorePartition] = None
+        self._store_transaction: Optional[PartitionTransaction] = None
+
+    def configure(
+        self,
+        topic: Topic,
+        producer: RowProducer,
+        *,
+        store_partition: StorePartition,
+        **kwargs,
```

Comment:
The type checker yells at this method:
```Signature of method 'StatefulSource. configure()' does not match signature of the base method in class 'BaseSource' ```

Reply:
The `store_partition` argument is required in `StatefulSource.configure`, which makes `BaseSource.configure` and `StatefulSource.configure` incompatible